src/Common/HashTable/FixedHashTable.h

```diff
@@ -294,36 +296,28 @@ class FixedHashTable : private boost::noncopyable, protected Allocator, protecte
 
     const_iterator begin() const
     {
-        if (!buf)
+        if (!buf && min > max)
```

Comment:
pls let's clarify what `min > max` means.
maybe it makes sense to extract it into a separate function with a self-explanatory name. afaiu it means that container is empty.

Reply:
Yes. In that case, `min > max` means the container is empty. Then outside `it` is always larger than the `end()` and the for loop will not stop.

```
    void ALWAYS_INLINE mergeToViaEmplace(Self & that, Func && func)
    {
        for (auto it = this->begin(), end = this->end(); it != end; ++it)
        {
            typename Self::LookupResult res_it;
            bool inserted;
            that.emplace(it->getKey(), res_it, inserted, it.getHash());
            func(res_it->getMapped(), it->getMapped(), inserted);
        }
    }
```